src/compile/mark/label.ts

```diff
@@ -0,0 +1,95 @@
+import {UnitModel} from '../unit';
+import {X, Y, COLOR, TEXT, SIZE, ANCHOR, OFFSET} from '../../channel';
+import {applyMarkConfig, applyColorAndOpacity, formatMixins} from '../common';
+import {extend, contains} from '../../util';
+import {QUANTITATIVE, ORDINAL, TEMPORAL} from '../../type';
+
+export namespace label {
+  export function markType() {
+    return 'text';
+  }
+
+  export function background(model: UnitModel) {
+    return {
+      x: { value: 0 },
+      y: { value: 0 },
+      width: { field: { group: 'width' } },
+      height: { field: { group: 'height' } },
+      fill: {
+        scale: model.scaleName(COLOR),
+        field: model.field(COLOR, model.fieldDef(COLOR).type === ORDINAL ? {prefn: 'rank_'} : {})
+      }
+    };
+  }
+
+  export function properties(model: UnitModel) {
+    // TODO Use Vega's marks properties interface
+    let p: any = {
+      xc: {field: 'label_xc'},
+      yc: {field: 'label_yc'},
+      align: {field: 'label_align'},
+      fill: {field: 'label_color'}
+    };
+
+    applyMarkConfig(p, model,
+      ['angle', 'baseline', 'dx', 'dy', 'font', 'fontWeight',
+        'fontStyle', 'radius', 'theta', 'text']);
+        
+    const fieldDef = model.fieldDef(TEXT);
+
+    // size
+    if (model.has(SIZE)) {
```

Comment:
split this into `size()` method and get rid of `sizeValue` method like other files.  (We just refactored this yesterday.)